src/activities/discussions/DiscussionTopicEntity.js

```diff
@@ -56,6 +56,99 @@ export class DiscussionTopicEntity extends Entity {
 		return true;
 	}
 
+	/**
+	 * @returns {bool} Whether or not the edit description action is present on the topic description entity
+	 */
+	canEditDescription() {
+		const descriptionEntity = this._getDescriptionEntity();
+		return descriptionEntity
+			&& descriptionEntity.hasActionByName(Actions.discussions.topic.updateDescription);
+	}
+
+	/**
+	 * @returns {object} a helper function to get topic description entity
+	 */
+
+	_getDescriptionEntity() {
+		return this._entity
+			&& this._entity.hasSubEntityByRel(Rels.Discussions.description)
+			&& this._entity.getSubEntityByRel(Rels.Discussions.description);
+	}
+
+	/**
+	 * @returns {string} Topic description in plaintext (HTML stripped)
+	 */
+	descriptionPlaintext() {
+		const descriptionEntity = this._getDescriptionEntity();
+		return descriptionEntity
+			&& descriptionEntity.properties
+			&& descriptionEntity.properties.text;
+	}
+
+	/**
+	 * @returns {string} Topic description in HTML
+	 */
+	descriptionHtml() {
+		const descriptionEntity = this._getDescriptionEntity();
+		if (!descriptionEntity || !descriptionEntity.properties || !descriptionEntity.properties.html) {
+			return;
+		}
+		return descriptionEntity.properties.html;
+	}
+
+	/**
+	 * @returns {string} Topic description formatted to be used with a d2l-html-editor
+	 */
+	descriptionEditorHtml() {
+		const descriptionEntity = this._getDescriptionEntity();
+		if (!descriptionEntity) {
+			return;
+		}
+
+		const updateDescriptionAction = descriptionEntity.getActionByName(Actions.discussions.topic.updateDescription);
+		return updateDescriptionAction
+			&& updateDescriptionAction.hasFieldByName('description')
+			&& updateDescriptionAction.getFieldByName('description').value;
+	}
+
+	/**
+	 * @returns {bool} A helper function, to see if description needs updating
+	 */
+
+	_hasDescriptionChanged(description) {
+		return description !== this.descriptionEditorHtml();
```

Comment:
Can you use `_hasFieldValueChanged` in the same file instead of creating a function for every property? I'm pretty sure you need to add description to the equals method as well

Reply:
This please!

Reply:
Sure, fixed it